File: cs/DressDiscover/DressDiscover.Api/src/DressDiscover/Api/Models/Worksheet/WorksheetFeatureSetId.cs
namespace DressDiscover.Api.Models.Worksheet
{
    public sealed class WorksheetFeatureSetId {
        public static WorksheetFeatureSetId Parse(string value) {
            if (value.Length == 0) {
                throw new System.ArgumentException("id is empty", "WorksheetFeatureSetId");
            }
            return new WorksheetFeatureSetId(value);
        }

        private WorksheetFeatureSetId(string value) {
            this.value = value;
        }

        public override string ToString() {
            return this.value;
        }

        private string value;
    }
}

File: worksheet/ts/src/ts/models/worksheet/worksheet_feature_set_id.ts
export class WorksheetFeatureSetId {
    public static parse(id: string): WorksheetFeatureSetId {
        if (id.trim().length === 0) {
            throw new RangeError("id is empty");
        }
        return new WorksheetFeatureSetId(id);
    }

    public equals(other?: WorksheetFeatureSetId): boolean {
        if (typeof(other) === "undefined") {
            return false;
        }
        return this.id === other.id;
    }

    public toString(): string {
        return this.id;
    }

    private constructor(private readonly id: string) {
    }
}
